diff -uNr pbbuttonsd-0.5.11-orig/libpbbipc/pbb.h pbbuttonsd-0.5.11/libpbbipc/pbb.h
--- pbbuttonsd-0.5.11-orig/libpbbipc/pbb.h	2004-02-09 19:29:00.000000000 +0100
+++ pbbuttonsd-0.5.11/libpbbipc/pbb.h	2004-05-02 21:38:49.787623784 +0200
@@ -32,6 +32,10 @@
 #define PBBDF_NONE	0   /* Flags for deamonise functions */
 #define PBBDF_FORCE	1
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 /* --- private module data structure of libpbbuttons --- */
 struct libbase {
 	int serverport;
@@ -65,4 +69,8 @@
 /* this function should not be used in new programs */
 void detach_process (char *name);
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif    /* INCLUDE_PBB_H */
diff -uNr pbbuttonsd-0.5.11-orig/libpbbipc/pbberrno.h pbbuttonsd-0.5.11/libpbbipc/pbberrno.h
--- pbbuttonsd-0.5.11-orig/libpbbipc/pbberrno.h	2003-11-04 06:00:31.000000000 +0100
+++ pbbuttonsd-0.5.11/libpbbipc/pbberrno.h	2004-05-02 21:39:41.225803992 +0200
@@ -37,7 +37,15 @@
 #define E_NOMEM		23
 #define E_NOSUPPORT	24
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 /* prototypes */
 void print_error(char *msg, ...);
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif    /* INCLUDE_PBBERRNO_H */
diff -uNr pbbuttonsd-0.5.11-orig/libpbbipc/pbbipc.h pbbuttonsd-0.5.11/libpbbipc/pbbipc.h
--- pbbuttonsd-0.5.11-orig/libpbbipc/pbbipc.h	2003-09-06 11:31:23.000000000 +0200
+++ pbbuttonsd-0.5.11/libpbbipc/pbbipc.h	2004-05-02 21:40:31.731126016 +0200
@@ -39,6 +39,10 @@
 #define WARNING				12		/* arg = taglist */
 #define READVALUE			13		/* arg = taglist */
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 struct pbbmessage {
 	long			messagetype;
 	int			returnport;
@@ -79,4 +83,8 @@
 /*debug */
 void peep_ipc (struct tagitem *taglist);
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif    /* INCLUDE_PBBIPC_H */
diff -uNr pbbuttonsd-0.5.11-orig/libpbbipc/pbbmisc.h pbbuttonsd-0.5.11/libpbbipc/pbbmisc.h
--- pbbuttonsd-0.5.11-orig/libpbbipc/pbbmisc.h	2004-02-09 19:48:09.000000000 +0100
+++ pbbuttonsd-0.5.11/libpbbipc/pbbmisc.h	2004-05-02 21:42:17.864991216 +0200
@@ -24,6 +24,10 @@
 #define TYPE_CHARDEV	2
 #define TYPE_SYMLINK	3
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 /* prototypes */
 int check_devorfile(char *file, int type);
 int launch_program (char* maskstr, ...);
@@ -37,4 +41,8 @@
 void cleanup_buffer(char *buffer);
 int axtoi (char *arg);
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif    /* INCLUDE_PBBMISC_H */
diff -uNr pbbuttonsd-0.5.11-orig/libpbbipc/pbbtaglist.h pbbuttonsd-0.5.11/libpbbipc/pbbtaglist.h
--- pbbuttonsd-0.5.11-orig/libpbbipc/pbbtaglist.h	2004-03-08 12:16:06.000000000 +0100
+++ pbbuttonsd-0.5.11/libpbbipc/pbbtaglist.h	2004-05-02 21:43:09.284174312 +0200
@@ -28,6 +28,10 @@
 #define SYSINFO_SLEEPSUPPORTED	0x04
 #define SYSINFO_PMUDSUPPORT	0x08
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 typedef unsigned long tag_t;
 
 struct tagitem {
@@ -44,4 +48,8 @@
 void taglist_add (struct tagitem *taglist, tag_t tag, tag_t val);
 int taglist_count (struct tagitem *taglist);
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif    /* INCLUDE_PBBTAGLIST_H */
