--- pbbuttonsd-0.5.6/libpbbipc/pbb.h.orig	2003-10-02 16:21:54.000000000 +0200
+++ pbbuttonsd-0.5.6/libpbbipc/pbb.h	2003-12-13 11:27:50.208345416 +0100
@@ -32,6 +32,10 @@
 #define PBBDF_NONE	0   /* Flags for deamonise functions */
 #define PBBDF_FORCE	1
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 /* --- private module data structure of libpbbuttons --- */
 struct libbase {
 	int serverport;
@@ -63,4 +67,8 @@
 /* this function should not be used in new programs */
 void detach_process (char *name);
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif    /* INCLUDE_PBB_H */
--- pbbuttonsd-0.5.6/libpbbipc/pbberrno.h.orig	2003-11-04 06:00:31.000000000 +0100
+++ pbbuttonsd-0.5.6/libpbbipc/pbberrno.h	2003-12-13 11:28:16.139403296 +0100
@@ -37,7 +37,15 @@
 #define E_NOMEM		23
 #define E_NOSUPPORT	24
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 /* prototypes */
 void print_error(char *msg, ...);
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif    /* INCLUDE_PBBERRNO_H */
--- pbbuttonsd-0.5.6/libpbbipc/pbbipc.h.orig	2003-09-06 11:31:23.000000000 +0200
+++ pbbuttonsd-0.5.6/libpbbipc/pbbipc.h	2003-12-13 11:28:44.010166296 +0100
@@ -39,6 +39,10 @@
 #define WARNING				12		/* arg = taglist */
 #define READVALUE			13		/* arg = taglist */
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 struct pbbmessage {
 	long			messagetype;
 	int			returnport;
@@ -79,4 +83,8 @@
 /*debug */
 void peep_ipc (struct tagitem *taglist);
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif    /* INCLUDE_PBBIPC_H */
--- pbbuttonsd-0.5.6/libpbbipc/pbbmisc.h.orig	2003-09-24 13:58:20.000000000 +0200
+++ pbbuttonsd-0.5.6/libpbbipc/pbbmisc.h	2003-12-13 11:29:32.511792928 +0100
@@ -25,13 +25,21 @@
 #define TYPE_CHARDEV	2
 #define TYPE_SYMLINK	3
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 /* prototypes */
 int check_devorfile(char *file, int type);
-int launch_program (char* template, ...);
+int launch_program (char* templ, ...);
 int check_permissions (char *file, int uid, int mask);
 int get_owner (char *file);
 int get_permissions (char *file);
 int keydelayms (struct timeval *start, int value, int delay);
 void cleanup_buffer(char *buffer);
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif    /* INCLUDE_PBBMISC_H */
--- pbbuttonsd-0.5.6/libpbbipc/pbbtaglist.h.orig	2003-09-30 13:31:27.000000000 +0200
+++ pbbuttonsd-0.5.6/libpbbipc/pbbtaglist.h	2003-12-13 11:29:56.767105560 +0100
@@ -21,6 +21,10 @@
 
 #define MASK_ERROR		~FLG_ERROR
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 typedef unsigned long tag_t;
 
 struct tagitem {
@@ -37,4 +41,8 @@
 void taglist_add (struct tagitem *taglist, tag_t tag, tag_t val);
 int taglist_count (struct tagitem *taglist);
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif    /* INCLUDE_PBBTAGLIST_H */
